consult.el

```diff
@@ -1102,46 +1102,81 @@ for which the command history is used."
    ;; If pressing "C-x M-:", i.e., `repeat-complex-command',
    ;; we are instead querying the `command-history' and get a full s-expression.
    ((eq last-command 'repeat-complex-command)
-    (delete-dups (mapcar #'prin1-to-string command-history)))
+    (cons (delete-dups (mapcar #'prin1-to-string command-history))
+          'minibuffer-history-position))
    ;; In the minibuffer we use the current minibuffer history,
    ;; which can be configured by setting `minibuffer-history-variable'.
    ((minibufferp)
-    (if (fboundp 'minibuffer-history-value)
-        (minibuffer-history-value) ;; Emacs 27 only
-      (symbol-value minibuffer-history-variable)))
+    (cons (if (fboundp 'minibuffer-history-value)
+              (minibuffer-history-value) ;; Emacs 27 only
+            (symbol-value minibuffer-history-variable))
+          'minibuffer-history-position))
    ;; Otherwise we use a mode-specific history, see `consult-mode-histories'.
-   (t (when-let (history
-                 (or (seq-find (lambda (ring)
-                                 (and (derived-mode-p (car ring))
-                                      (boundp (cdr ring))))
-                               consult-mode-histories)
-                     (user-error
-                      "No history configured for `%s', see `consult-mode-histories'"
-                      major-mode)))
-        (symbol-value (cdr history))))))
+   (t (pcase-let ((`(,history ,index-var)
+                   (or (cdr (seq-find (lambda (ring)
+                                        (and (derived-mode-p (car ring))
+                                             (boundp (cadr ring))))
+                                      consult-mode-histories))
+                       (user-error
+                        "No history configured for `%s', see `consult-mode-histories'"
+                        major-mode))))
+        (cons (symbol-value history)
+              index-var)))))
+
+(defun consult--label-and-delete-dups (entries)
```

Comment:
This function is unnecessary. See how it is done for consult-kmacro. There I am using seq-uniq.

Reply:
But maybe you want to avoid the O(n^2) behavior which is a good idea. Is it justified here? Probably yes when the histories become very long? But then the function should be written in a more general form, e.g., `consult--unique-by` which takes a function argument to extract the element to check for equality or `consult--unique-car` if you want to check the car of an alist for equality.

Reply:
Now that I think about it, we should indeed write such a consult--unique-by-car function which can then also be used by consult-kmacro instead of the quadratic seq-uniq. This would be good to have!

Reply:
OK, I've added this function and made it work (I think) with `consult--kmacro`.